Makefile

```diff
@@ -61,29 +61,11 @@ E2E_FOCUS := "functional tests"
 
 # Directories.
 TOOLS_DIR := hack/tools
-TOOLS_BIN_DIR := $(TOOLS_DIR)/bin
+TOOLS_BIN_DIR := $(abspath $(TOOLS_DIR)/bin)
 BIN_DIR := bin
 TEST_DIR := test
 TEST_E2E_DIR := $(TEST_DIR)/e2e
-ENVSUBST_BIN := bin/envsubst
-ENVSUBST := $(TOOLS_DIR)/$(ENVSUBST_BIN)
-
-# Binaries.
-KUSTOMIZE := $(abspath $(TOOLS_BIN_DIR)/kustomize)
-CONTROLLER_GEN := $(abspath $(TOOLS_BIN_DIR)/controller-gen)
-ENVSUBST := $(abspath $(TOOLS_BIN_DIR)/envsubst)
-
-envsubst: $(ENVSUBST) ## Build a local copy of envsubst.
-$(ENVSUBST): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR) && go build -tags=tools -o $(ENVSUBST_BIN) github.com/drone/envsubst/cmd/envsubst
-
-controller-gen: $(CONTROLLER_GEN) ## Build a local copy of controller-gen.
-$(CONTROLLER_GEN): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR) && go build -tags=tools -o $(BIN_DIR)/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
-
-kustomize: $(KUSTOMIZE)
-$(KUSTOMIZE): $(TOOLS_DIR)/go.mod # Build kustomize from tools folder.
-	cd $(TOOLS_DIR); go build -tags=tools -o $(BIN_DIR)/kustomize sigs.k8s.io/kustomize/kustomize/v3
+GO_INSTALL = ./scripts/go_install.sh
```

Comment:
Is this custom-made or is this pattern used in some other projects?

Reply:
this is a pattern used in other cluster-api-providers projects and also in other k8s projects